apps/financial-aid/web-osk/src/routes/application/RelationshipstatusForm/RelationshipstatusForm.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState, useContext } from 'react'
+import { Text } from '@island.is/island-ui/core'
+
+import {
+  ContentContainer,
+  Footer,
+  RadioButtonContainer,
+  SpouseInfo,
+} from '@island.is/financial-aid-web/osk/src/components'
+
+import { useRouter } from 'next/router'
+
+import useFormNavigation from '@island.is/financial-aid-web/osk/src/utils/useFormNavigation'
+import cn from 'classnames'
+import * as styles from './relationshipstatusForm.treat'
+
+import {
+  FamilyStatus,
+  isEmailValid,
+  NavigationProps,
+} from '@island.is/financial-aid/shared/lib'
+import { UserContext } from '@island.is/financial-aid-web/osk/src/components/UserProvider/UserProvider'
+
+const RelationshipstatusForm = () => {
+  const router = useRouter()
+
+  const { user, setUser } = useContext(UserContext)
+
+  const navigation: NavigationProps = useFormNavigation(
+    router.pathname,
+  ) as NavigationProps
+
+  const [hasError, setHasError] = useState(false)
+
+  const [acceptData, setAcceptData] = useState(false)
+
+  const options = [
+    {
+      label: 'Nei, ég er ekki í sambúð',
+      value: FamilyStatus.UNKNOWN,
+    },
+    {
+      label: 'Já, ég er í óstaðfestri sambúð',
+      value: FamilyStatus.UNREGISTEREDCOBAHITATION,
+    },
+  ]
+
+  const errorCheck = () => {
+    if (!user?.familyStatus) {
+      setHasError(true)
+      return
+    }
+
+    if (!navigation?.nextUrl) {
+      return
+    }
+
+    if (user?.familyStatus === FamilyStatus.UNKNOWN) {
+      router.push(navigation?.nextUrl)
+    }
+
+    if (user?.familyStatus === FamilyStatus.UNREGISTEREDCOBAHITATION) {
+      if (!acceptData || !user.spouse?.email || !user.spouse?.nationalId) {
+        setHasError(true)
+        return
+      }
+
+      if (
+        !isEmailValid(user.spouse?.email) ||
+        user.spouse?.nationalId.length !== 10
+      ) {
+        setHasError(true)
+        return
+      }
```

Comment:
I would join these if statements and drag the boolean value for the if into a const